examples/md-flexible/scripts/plotPerTraversalAndParameter.py

```diff
@@ -0,0 +1,253 @@
+                currentLine.strip()
+                densityPerTraversal[allTraversals.index(traversal)].append(float(currentLine))
+            elif density and (match := re.search(regexNumOfParticlesAbsolute, line)) is not None:
+                currentLine = line.split(':', 1)[1]
+                currentLine.strip()
+                currentDensity = float(currentLine)
+            elif (match := re.search(regexIterTook, line)) is not None:
+                values.append(int(match.group(1)))  # append time needed to perform iteration
+        foundTraversal = False
+
+    timePerTravsersal[allTraversals.index(traversal)].append((sum(values)/len(values)))  # append mean time of this traversal
+    if size or density:
+        boxSize = numpy.prod([a - b for a, b in zip(boxSizeListMax, boxSizeListMin)])  # get size per dimension and then multiply to get overall boxsize
+        boxSizePerTraversal[allTraversals.index(traversal)].append(boxSize)
+        if density:
+            currentDensity = currentDensity / boxSize
+            densityPerTraversal[allTraversals.index(traversal)].append(currentDensity)
+
+
+# ---------------------------------------------- Functions ----------------------------------------
+
+def getContainerByTraversal(traversal):
+    for c in allContainers:
+        if c.__contains__(traversal):
+            return c
+
+def calculateMeans(xAxis, yAxis):
+    global meanYaxis
+    global meanXaxis
+    global valuesErrorMinus
+    global valuesErrorPlus
+    meanXaxis = []
+    meanYaxis = []
+    valuesErrorMinus = []
+    valuesErrorPlus = []
+    counter = 0
+    for x in xAxis:
+        if meanXaxis.__contains__(x):
+            newIndex = meanXaxis.index(x)
+            oldIndex = counter
+            meanYaxis[newIndex] += yAxis[oldIndex]
+            if valuesErrorMinus[newIndex] > yAxis[oldIndex]:
+                valuesErrorMinus[newIndex] = yAxis[oldIndex]
+            if valuesErrorPlus[newIndex] < yAxis[oldIndex]:
+                valuesErrorPlus[newIndex] = yAxis[oldIndex]
+            counter += 1
+        else:
+            # append x in x-axis and y ind y-axis as well as for the highest and lowest y value so far
+            meanXaxis.append(x)
+            meanYaxis.append(yAxis[xAxis.index(x)])
+            valuesErrorMinus.append(yAxis[xAxis.index(x)])
+            valuesErrorPlus.append(yAxis[xAxis.index(x)])
+            counter += 1
+    divisor = len(yAxis) / len(meanYaxis)
+    valuesErrorMinus.sort()
+    valuesErrorPlus.sort()
+    meanYaxis.sort()
+    meanYaxis = [y / divisor for y in meanYaxis]
+    meanXaxis.sort()
+
+    # TODO: sort depending
+
+# ---------------------------------------------- Plot ---------------------------------------------
+
+# build the full rgb color space
+colorrange = ['rgb(  0, 0, 0)', 'rgb(0, 100,   100)', 'rgb(255, 219,   0)', 'rgb(255, 0,   0)', 'rgb(  0, 147, 255)', 'rgb(71, 0, 255)', 'rgb(73, 255,   0)', 'rgb(255,   0, 221)']
```

Comment:
the comment does not fit anymore ;)
I think it would be nicer if the number of colors is generated so that even if we add more containers there will still be enough colors.